packages/sources/alphavantage-test/src/endpoint/forex.ts

```diff
@@ -0,0 +1,51 @@
+import { EndpointTypes } from './router'
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { makeLogger } from '@chainlink/external-adapter-framework/util'
+
+const logger = makeLogger('Alphavantage HttpEndpoint')
+
+export const httpTransport = new HttpTransport<EndpointTypes>({
+  prepareRequests: (params, config) => {
+    return params.map((param) => {
+      const from = param.base
+      const to = param.quote
+
+      const requestConfig = {
+        baseURL: config.API_ENDPOINT,
+        method: 'GET',
+        params: {
+          function: 'CURRENCY_EXCHANGE_RATE',
+          from_currency: from,
+          to_currency: to,
+          from_symbol: from,
+          to_symbol: to,
+          symbol: from,
+          market: to,
+          apikey: config.API_KEY,
+        },
+      }
+      return {
+        params,
+        request: requestConfig,
+      }
+    })
+  },
+  parseResponse: (params, res) => {
+    if (!res.data) {
+      logger.error(`There was a problem getting the data from the source`)
+      return []
+    }
+    return params.map((param) => {
+      const result = Number(res.data['Realtime Currency Exchange Rate']['5. Exchange Rate'])
```

Comment:
We should add some validation that checks whether the response has a value for `Realtime Currency Exchange Rate` and `5. Exchange Rate` separately so we could return an error response if one doesn't.

Reply:
We could also add a check if `Error Message` returned anything

Reply:
Done